encrypted-media/scripts/generate-request-disallowed-input.js

```diff
@@ -1,92 +1,74 @@
+        0x10, 0x77, 0xEF, 0xEC, 0xC0, 0xB2, 0x4D, 0x02,  // Common SystemID
+        0xAC, 0xE3, 0x3C, 0x1E, 0x52, 0xE2, 0xFB, 0x4B,
+        0x00, 0x00, 0x00, 0x00                           // datasize
+    ]);
+    push_test(config.keysystem, 'cenc', initData, testnamePrefix( qualifier, config.keysystem ) + ', temporary, cenc, invalid initdata (not pssh)');
 
-    promise_test(function()
-    {
-        // Invalid data as type = 'psss'.
-        var initData = new Uint8Array([
-            0x00, 0x00, 0x00, 0x00,                          // size = 0
-            0x70, 0x73, 0x73, 0x73,                          // 'psss'
-            0x00,                                            // version = 0
-            0x00, 0x00, 0x00,                                // flags
-            0x10, 0x77, 0xEF, 0xEC, 0xC0, 0xB2, 0x4D, 0x02,  // Common SystemID
-            0xAC, 0xE3, 0x3C, 0x1E, 0x52, 0xE2, 0xFB, 0x4B,
-            0x00, 0x00, 0x00, 0x00                           // datasize
-        ]);
-        return test_session(config.keysystem, 'cenc', initData);
-    }, testnamePrefix( qualifier, config.keysystem ) + ', temporary, cenc, invalid initdata (not pssh)');
+    // Valid key ID size must be at least 1 character for keyids.
+    keyId = new Uint8Array(0);
+    initData = stringToUint8Array(createKeyIDs(keyId));
+    push_test(config.keysystem, 'keyids', initData, testnamePrefix( qualifier, config.keysystem ) + ', temporary, keyids, invalid initdata (too short key ID)');
 
-    promise_test(function()
-    {
-        // Valid key ID size must be at least 1 character for keyids.
-        var keyId = new Uint8Array(0);
-        var initData = stringToUint8Array(createKeyIDs(keyId));
-        return test_session(config.keysystem, 'keyids', initData);
-    }, testnamePrefix( qualifier, config.keysystem ) + ', temporary, keyids, invalid initdata (too short key ID)');
+    // Valid key ID size must be less than 512 characters for keyids.
+    keyId = new Uint8Array(600);
+    initData = stringToUint8Array(createKeyIDs(keyId));
+    push_test(config.keysystem, 'keyids', initData, testnamePrefix( qualifier, config.keysystem ) + ', temporary, keyids, invalid initdata (too long key ID)');
 
-    promise_test(function()
-    {
-        // Valid key ID size must be less than 512 characters for keyids.
-        var keyId = new Uint8Array(600);
-        var initData = stringToUint8Array(createKeyIDs(keyId));
-        return test_session(config.keysystem, 'keyids', initData);
-    }, testnamePrefix( qualifier, config.keysystem ) + ', temporary, keyids, invalid initdata (too long key ID)');
+    Promise.all( tests.map( function( testspec ) {
+        return isInitDataTypeSupported(testspec.keysystem,testspec.initDataType);
+    })).then( function( results ) {
+        tests.filter( function( testspec, i ) { return results[ i ]; } ).forEach( function( testspec ) {
+            async_test( function( test ) {
```

Comment:
Why promise_tests were changed to async_tests? Can [promise_rejects](https://github.com/w3c/testharness.js/blob/master/docs/api.md#promise-tests) be used here?